File: app/src/main/java/br/com/nicetuber/modules/ApiModule.java
package br.com.nicetuber.modules;

import javax.inject.Singleton;

import br.com.nicetuber.model.api.YoutubeApi;
import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

/**
 * Created by ismael on 3/20/17.
 */
@Module
public class ApiModule {

    @Provides
    @Singleton
    public YoutubeApi providesProductApi(Retrofit retrofit) {
        return retrofit.create(YoutubeApi.class);
    }

}

File: app/src/main/java/br/com/nicetuber/modules/ApiModule.kt
package br.com.nicetuber.modules

import br.com.nicetuber.model.api.YoutubeApi
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import javax.inject.Singleton

/**
 * Created by ismael on 3/20/17.
 */
@Module
class ApiModule {

    @Provides
    @Singleton
    fun providesYoutubeApi(retrofit: Retrofit): YoutubeApi {
        return retrofit.create(YoutubeApi::class.java)
    }

}
